lara-typescript/src/interactive-api-client/types.ts

```diff
@@ -308,13 +308,20 @@ export interface IDecoratedContentEvent {
 }
 
 export interface ITextDecorationInfo {
+  words: string[];
+  replace: string;
+  wordClass: string;
+  listenerTypes: Array<{type: string}>;
+}
+
+export interface ITextDecorationHandlerInfo {
   words: string[];
   replace: string;
   wordClass: string;
   eventListeners: IEventListeners;
 }
```

Comment:
You could DRY up the above:
```typescript
interface ITextDecorationBaseInfo {
  words: string[];
  replace: string;
  wordClass: string;
}

export interface ITextDecorationInfo extends ITextDecorationBaseInfo {
  listenerTypes: Array<{type: string}>;
}

export interface ITextDecorationHandlerInfo extends ITextDecorationBaseInfo {
  eventListeners: IEventListeners;
}
```
Also, I'm wondering if there might be more descriptive naming options -- maybe `TextDecorationStaticInfo` and `TextDecorationRuntimeInfo`? Just throwing some options out there to get the creative juices flowing.

Reply:
Finally, I think you could still call the field `listeners` in both cases -- the interpretation would just be that it's the actual listener functions in one case and static descriptions of the listeners (currently including only the type but potentially extensible to other information down the road).

Reply:
I added the DRY suggestions.  Still not certain on names.  Let's think it over some more.